h/static/scripts/base/controller.js

```diff
@@ -7,21 +7,22 @@
  * Multiple controllers may need to refer to the same element, so `data-ref`
  * supports a space-separated list of reference names.
  */
-function findRefs(el, map) {
-  if (!el.dataset) {
-    return map;
-  }
-  map = map || {};
+function findRefs(el) {
+  var map = {};
 
-  var refs = (el.dataset.ref || '').split(' ');
+  // Use `Element#getAttribute` rather than `Element#dataset` to support IE 10
+  // and avoid https://bugs.webkit.org/show_bug.cgi?id=161454
+  var refs = (el.getAttribute('data-ref') || '').split(' ');
   refs.forEach(function (ref) {
     map[ref] = el;
   });
 
-  for (var i=0; i < el.children.length; i++) {
-    var node = el.children[i];
+  // Use `Node#childNodes` rather than `ParentNode#children` because
+  // IE lacks support for `#children` on <svg> elements
+  for (var i=0; i < el.childNodes.length; i++) {
+    var node = el.childNodes[i];
     if (node.nodeType === Node.ELEMENT_NODE) {
-      findRefs(node, map);
+      Object.assign(map, findRefs(node));
```

Comment:
I'm not sure I follow. Surely you can use `el.querySelectorAll` on a detached element:

```javascript
var el = document.createElement('div');  # => <div></div>
el.innerHTML = '<span id="a" data-ref="foo"><span id="b" data-ref="bar"></span></span><span id="c" data-ref="baz"></span>'
el.querySelectorAll('*[data-ref]'); # => [span#a, span#b, span#c]
```